legacy/toml-config/src/main/java/io/zeebe/legacy/tomlconfig/LegacyConfigurationSupport.java

```diff
@@ -0,0 +1,257 @@
+                          insertionPoint); // "gateway.*"
+
+              MAPPING_BROKER.put(
+                  oldEnvironmentVariable, new Replacement(newEnvironmentVariableInBrokerContext));
+            });
+    VALUE_CONVERTERS_BROKER.putAll(VALUE_CONVERTERS_GATEWAY);
+
+    // zeebe-broker.network
+    MAPPING_BROKER.put("ZEEBE_HOST", new Replacement("zeebe-broker.network.host"));
+    MAPPING_BROKER.put(
+        "ZEEBE_ADVERTISED_HOST", new Replacement("zeebe-broker.network.advertisedHost"));
+    MAPPING_BROKER.put("ZEEBE_PORT_OFFSET", new Replacement("zeebe-broker.network.portOffset"));
+
+    // TODO check docker files and shell scripts and change those as well;
+    //  if so remove hint in header of sample config files
+    //  also test that it works when using the startup script
+
+  }
+
+  private final Scope scope;
+
+  public LegacyConfigurationSupport(Scope scope) {
+    this.scope = scope;
+  }
+
+  /**
+   * This method checks whether the program was called with a toml configuration file. If so, it
+   * prints out a warning.
+   */
+  public void checkForLegacyTomlConfigurationArgument(String[] args, String recommendedSetting) {
+    if (args.length == 1 && args[0].endsWith(".toml")) {
+      final String configFileArgument = args[0];
+      Loggers.LEGACY_LOGGER.warn(
+          "Found command line argument "
+              + configFileArgument
+              + " which might be a TOML configuration file.");
+      Loggers.LEGACY_LOGGER.info(
+          "TOML configuration files are no longer supported. Please specify a YAML configuration file"
+              + "and set it via environment variable \"spring.config.additional-location\" (e.g. "
+              + "\"export spring.config.additional-location='file:./config/"
+              + recommendedSetting
+              + "'\").");
+      Loggers.LEGACY_LOGGER.info(
+          "The ./config/ folder contains a configuration file template. Alternatively, you can also use environment variables.");
+    }
+  }
+
+  /**
+   * This method checks for legacy environment variables. If it finds an old environment variable
+   * which is set, it looks whether the new counterpart is set. If the new counterpart is set, it
+   * does nothing. If it is not set, it sets a system setting under the new key that has the same
+   * value as is associated with the old environment variable. This effectively makes the old
+   * environment variable value visible under the new environment variable. It also prints out a
+   * warning to the log.
+   */
+  public void checkForLegacyEnvironmentVariables() {
+
+    final Map<String, Replacement> mappingTable;
+    final Map<String, Function<String, String>> valueConverters;
+
+    switch (scope) {
+      case GATEWAY:
+        mappingTable = MAPPING_GATEWAY;
+        valueConverters = VALUE_CONVERTERS_GATEWAY;
+        break;
+
+      case BROKER:
+        mappingTable = MAPPING_BROKER;
+        valueConverters = Collections.emptyMap();
```

Comment:
I guess we meant to use `VALUE_CONVERTERS_BROKER` here?

Reply:
done